### PR TITLE
Remove cleanup function

### PR DESCRIPTION
**Pull request by Emantor**
This removes the cleanup function and uses on_deactivate instead.
Do we deactivate the targets drivers after a successful test?

### CODE REVIEW

File: labgrid/target.py [hunk 1]
```diff
@@ -203,12 +203,3 @@ def deactivate(self, client):
...
         client.state = BindingState.bound
-
-    def cleanup(self):
-        """Clean up conntected drivers and resources in reversed order"""
-        for drv in reversed(self.drivers):
-            if hasattr(drv, 'cleanup') and callable(getattr(drv, 'cleanup')):
-                drv.cleanup()
-        for res in reversed(self.resources):
-            if hasattr(res, 'cleanup') and callable(getattr(res, 'cleanup')):
-                res.cleanup()
```
Comment (jluebbe) [inline labgrid/target.py]: This should be kept and deactivate the drivers instead.

Comment (jluebbe) [inline labgrid/target.py]: This should be kept and deactivate the drivers instead.

File: labgrid/environment.py [hunk 2]
```diff
@@ -33,7 +33,3 @@ def __attrs_post_init__(self):
     def get_target(self, role: str='main') -> Target:
         """Returns the specified target."""
         return self.targets[role]
-
-    def cleanup(self):
```
Comment (jluebbe) [inline labgrid/environment.py]: This should stay, as the cleanup function on drivers should stay as well (see below).

**Comment by Emantor** [comment]
Pushed a new version, pls review again.

**Comment by jluebbe** [comment]
Looks good now. I'll wait for some test results before merging this.

**Comment by Emantor** [comment]
I ran my tests overnight, SSHDriver cleanup function was not called before and throws an error now after the test ends, I'll fix that soon™.

**Pull request merged**